systems/trajectory_optimization/dircon/dircon.cc

```diff
@@ -102,6 +103,11 @@ Dircon<T>::Dircon(std::unique_ptr<DirconModeSequence<T>> my_sequence,
     if (i_mode > 0) {
       v_post_impact_vars_.push_back(NewContinuousVariables(
           plant_.num_velocities(), "v_p[" + std::to_string(i_mode) + "]"));
+      v_post_impact_vars_substitute_.push_back(
+          {state().tail(plant_.num_velocities()),
+           Eigen::Map<MatrixXDecisionVariable>(
+               v_post_impact_vars_.back().data(), plant_.num_velocities(), 1)
+               .cast<Expression>()});
```

Comment:
Why use Eigen::Map here? An alternate option would be 
```
v_post_impact_vars_.back().tail(plant_.num_velocities()).template cast<Expression>()}
```

It's a bit easier to read, though without the memory efficiency of Map.

Reply:
I was using https://github.com/RobotLocomotion/drake/blob/a3312d0f2d526c074f793c0c7eba7f41153c7055/systems/trajectory_optimization/multiple_shooting.cc#L95 as a reference. Though I'm not confident if `template_cast` wouldn't achieve the same effect.